sycl/test/esimd/on-device/PrefixSum.cpp

```diff
@@ -0,0 +1,470 @@
+//==----------- PrefixSum.cpp  - DPC++ ESIMD on-device test ----------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// TODO enable on Windows and Level Zero
+// REQUIRES: linux && gpu && opencl
+// RUN: %clangxx-esimd -fsycl %s -o %t.out
+// RUN: env SYCL_DEVICE_TYPE=HOST %t.out
+// RUN: %ESIMD_RUN_PLACEHOLDER %t.out
+
+#include "esimd_test_utils.hpp"
+
+#include <CL/sycl.hpp>
+#include <CL/sycl/INTEL/esimd.hpp>
+#include <iostream>
+
+#define MAX_TS_WIDTH 1024
+#define TUPLE_SZ 1 // kernel can only handle TUPLE_SZ can be 1, 2, 4
+
+#if TUPLE_SZ == 1
+#define GATHER_SCATTER_MASK ESIMD_R_ENABLE
+#elif TUPLE_SZ == 2
+#define GATHER_SCATTER_MASK ESIMD_GR_ENABLE
+#elif TUPLE_SZ == 4
+#define GATHER_SCATTER_MASK ESIMD_ABGR_ENABLE
+#endif
```

Comment:
Should we also have `RUN`'s for the test with different `TUPLE_SZ`?

Reply:
This is a demo case for implementing PrefixSum in ESIMD, with configurable parameters for tuning purpose. Some different TUPLE_SIZE are exercised in other variants of this kernel.